pkg/aws/cloudmap.go

```diff
@@ -0,0 +1,319 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/klog"
+
+	awssdk "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/appmesh"
+	"github.com/aws/aws-sdk-go/service/servicediscovery"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	//AttrAwsInstanceIPV4 is a special attribute expected by CloudMap.
+	//See https://github.com/aws/aws-sdk-go/blob/fd304fe4cb2ea1027e7fc7e21062beb768915fcc/service/servicediscovery/api.go#L5161
+	AttrAwsInstanceIPV4 = "AWS_INSTANCE_IPV4"
+	//AttrK8sPod is a custom attribute injected by app-mesh controller
+	AttrK8sPod = "k8s.io/pod"
+	//AttrK8sNamespace is a custom attribute injected by app-mesh controller
+	AttrK8sNamespace = "k8s.io/namespace"
+)
```

Comment:
https://docs.aws.amazon.com/AmazonECS/latest/developerguide/service-discovery.html

:+1: for REGION and AVAILABILITY_ZONE. This will be helpful for AZ-aware routing: aws/aws-app-mesh-roadmap#94

Reply:
Yes they are useful. Should we look into adding them as labels to pods and let register instance take over from there?

Reply:
I can see those labels being useful for pods--how does the cluster admin decide which pods get them?

Reply:
For now we can lookup the node corresponding to pod and get that information. Below is the information from node. However, controller needs permissions to get node info. We can follow up this work in upcoming versions.

```
labels:
    alpha.eksctl.io/cluster-name: cluster-20190821
    alpha.eksctl.io/instance-id: i-08dd903c3cbe5db66
    alpha.eksctl.io/nodegroup-name: m5-l-1
    beta.kubernetes.io/arch: amd64
    beta.kubernetes.io/instance-type: m5.large
    beta.kubernetes.io/os: linux
    failure-domain.beta.kubernetes.io/region: us-west-2
    failure-domain.beta.kubernetes.io/zone: us-west-2a
    kubernetes.io/hostname: ip-192-168-118-179.us-west-2.compute.internal
    role: workers
```